{{cookiecutter.app_name}}/{{cookiecutter.app_name}}/public/views.py

```diff
@@ -75,3 +83,97 @@ def about():
     """About page."""
     form = LoginForm(request.form)
     return render_template("public/about.html", form=form)
+{% if cookiecutter.use_google_logging == "yes" %}
+
+def get_google_provider_cfg():
+    """Returns Google Provide configuration."""
+    return requests.get(current_app.config["GOOGLE_DISCOVERY_URL"]).json()
+
+
+@blueprint.route("/login")
+def login():
+    """Google Login view."""
+    client = WebApplicationClient(current_app.config["GOOGLE_CLIENT_ID"])
+    # Find out what URL to hit for Google login
+    google_provider_cfg = get_google_provider_cfg()
+    authorization_endpoint = google_provider_cfg["authorization_endpoint"]
+
+    # Use library to construct the request for Google login and provide
+    # scopes that let you retrieve user's profile from Google
+    request_uri = client.prepare_request_uri(
+        authorization_endpoint,
+        redirect_uri=request.base_url + "/callback",
+        scope=["openid", "email", "profile"],
+    )
+    return redirect(request_uri)
+
+
+@blueprint.route("/login/callback")
+def callback():
+    """Google Login redirect endpoint."""
+    client = WebApplicationClient(current_app.config["GOOGLE_CLIENT_ID"])
+    # Get authorization code Google sent back to you
+    code = request.args.get("code")
+
+    # Find out what URL to hit to get tokens that allow you to ask for
+    # things on behalf of a user
+    google_provider_cfg = get_google_provider_cfg()
+    token_endpoint = google_provider_cfg["token_endpoint"]
+
+    # Prepare and send request to get tokens! Yay tokens!
+    token_url, headers, body = client.prepare_token_request(
+        token_endpoint,
+        authorization_response=request.url,
+        redirect_url=request.base_url,
+        code=code,
+    )
+    token_response = requests.post(
+        token_url,
+        headers=headers,
+        data=body,
+        auth=(
+            current_app.config["GOOGLE_CLIENT_ID"],
+            current_app.config["GOOGLE_CLIENT_SECRET"],
+        ),
+    )
+
+    # Parse the tokens!
+    client.parse_request_body_response(json.dumps(token_response.json()))
+
+    # Now that we have tokens (yay) let's find and hit URL
+    # from Google that gives you user's profile information,
+    # including their Google Profile Image and Email
+    userinfo_endpoint = google_provider_cfg["userinfo_endpoint"]
+    uri, headers, body = client.add_token(userinfo_endpoint)
+    userinfo_response = requests.get(uri, headers=headers, data=body)
+
+    # We want to make sure their email is verified.
+    # The user authenticated with Google, authorized our
+    # app, and now we've verified their email through Google!
+    if userinfo_response.json().get("email_verified"):
+        users_email = userinfo_response.json()["email"]
+        given_name = userinfo_response.json()["given_name"]
+        family_name = userinfo_response.json()["family_name"]
+        name = userinfo_response.json()["name"]
```

Comment:
Can the conversion to json be factored out so that it isn't called multiple times? Also, would it be safer to `.get` these keys?